### ISSUE TITLE
oneOfWeighted types only accept [ number, Maker<V> ]

### ISSUE THREAD

**Issue by ctholho** [oftherivier/fictional#42]
Using fictional in a project 

```
import { oneOfWeighted } from "fictional"

const test = oneOfWeighted("seed", [
  [0.2, "20 percent"],
  [null, "rest percent"]
])
```

This will throw a type error: Type 'null' is not assignable to type 'number'. [2322]
<img width="907" alt="Screenshot 2024-12-19 at 15 04 33" src="https://github.com/user-attachments/assets/f19d145f-6c79-49bb-af65-c5a11acb7c57" />

According to the [README](https://github.com/oftherivier/fictional?tab=readme-ov-file#oneOfWeighted), `null` should be acceptable here.

**Comment by justinvdm** [comment]
Hi @ctholho, I'm sorry for the very long delay and radio silence, and thank you for fixing! I'll release this later today and let you know.

**Comment by justinvdm** [comment]
@ctholho released as v2.1.1, and copycat bumped accordingly to v5.1.1 with the dep range updated for the new fictional version
